homeassistant/components/splunk/__init__.py

```diff
@@ -55,45 +55,47 @@
 
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
-    """Set up the Splunk component."""
-    conf = config[DOMAIN]
-    host = conf.get(CONF_HOST)
-    port = conf.get(CONF_PORT)
-    token = conf.get(CONF_TOKEN)
-    use_ssl = conf[CONF_SSL]
-    verify_ssl = conf.get(CONF_VERIFY_SSL)
-    name = conf.get(CONF_NAME)
-    entity_filter = conf[CONF_FILTER]
-
-    event_collector = hass_splunk(
+    """Old setup for Splunk component."""
+    if DOMAIN in config:
+        # Entity filters are not configurable in Config Flow, so are removed
+        del config[DOMAIN]["filter"]
```

Comment:
We should keep the filter in YAML as that's something we allow for data exporter integrations.

Reply:
To clarify, if there is a filter, I do not import it to config flow since I don't believe its possible to do filters with a config entry?

Reply:
Yes, don't import the filter, but import the rest of the YAML that represents the connection.

Reply:
I thought that is what I was already doing, but I see using `del` is probably the problem.

Reply:
The filter should be applied, we shouldn't remove it.

Reply:
I can remove it in 6 months though right? It's a fairly useless feature in this integration that no one I've spoken to uses.

Reply:
No, I don't think so. I think we should keep the filter as a YAML only feature.